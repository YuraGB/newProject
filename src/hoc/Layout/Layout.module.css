.Main {
    max-width: 210mm;
    margin: 1vh auto;
    box-sizing: border-box;
    box-shadow: 1px 1px 10px black;
    background-color: white;
    min-height: 90vh;
    display: grid;
    grid-template-areas: "toolbar content";
    grid-template-rows: 1fr;
    grid-template-columns: 1fr 1fr;
}

/*Mobile*/

@media (max-width: 640px) {
    .Main {
        grid-template-areas: "content";
        grid-template-columns: 100%;
        margin: 0;
    }

    .print_wrapper {
        display: none;
    }
}

@media (min-width: 768px) {
    main.Main {
        max-width: 210mm;
        margin: 1vh auto;
        box-sizing: border-box;
        box-shadow: 1px 1px 10px black;
        background-color: white;
        min-height: 90vh;

        display: grid;
        grid-template-areas: "toolbar content";
        grid-template-rows: 1fr;
        grid-template-columns: 40% 1fr;
    }

    .to_print {
        display: inline;
        margin: auto;
        width: auto;
        padding: 5px 20px;
        background-color: #fff;
        border-radius: 5px;
        color: grey;
        text-decoration: none;
        box-shadow: 1px 1px 5px black;
    }

    .print_wrapper {
        text-align: left;
        position: sticky;
        width: 100%;
        padding: 15px;
        top: 0;
        display: inline;
        z-index: 5;
    }
}

/*print strange behavior*/
@media (max-width: 300px) {
    .Main {
        grid-template-areas: "toolbar content";
        grid-template-rows: 1fr;
        grid-template-columns: 40% 1fr;
    }
}

@media print { html, body { height: initial !important; overflow: initial !important; }}