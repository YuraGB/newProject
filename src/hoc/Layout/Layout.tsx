/**
 * The Layout component
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */

import React from 'react';

import Auxx from "../Auxx/Auxx";
import classes from './Layout.module.css';
import LeftToolbar from "../../components/LeftToolbar/LeftToolbar";

/**
 * Layout hire order component
 *
 * @param props
 * @author Yurii Huriianov <yuhur1985@gmail.com
 */
const Layout:React.FC = (props): JSX.Element => {
    return (
        <Auxx>
            <main className={classes.Main}>
                <LeftToolbar/>
                {props.children}
            </main>
        </Auxx>
    )
};

export default Layout;
