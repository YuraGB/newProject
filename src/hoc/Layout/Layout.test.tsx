import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Layout from "./Layout";

describe("Blocks", () => {
  const blocks = {
    Languages: "",
    skills: { test: "test" },
    social_links: { test: "test" },
    Education: {
      title: "",
      url: "",
      description: {
        title: "",
        url: "",
      },
    },
    about_me: "",
    profile: {
      first_name: "Yurii",
      last_name: "Hurianov",
      sub_title: "",
    },
    contact_info: "",
    work_experience: {
      test: {
        duties: [""],
        projects: [""],
      },
    },
  };

  test("Content wouldn't be rendered", () => {
    // @ts-ignore
    render(<Layout blocks={null} />);
    expect(screen.queryByText("Yurii Hurianov")).toBeNull();
  });

  test("Content will be rendered", () => {
    // @ts-ignore
    render(<Layout blocks={blocks} />);
    expect(screen.queryByText("Yurii Hurianov")).toBeInTheDocument();
  });
});
