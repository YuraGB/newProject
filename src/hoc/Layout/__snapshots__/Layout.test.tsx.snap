// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests match snapshots 1`] = `
Array [
  <div
    className="print_wrapper"
    onClick={[Function]}
  >
    <a
      className="to_print"
      href="#"
    >
      Print this out!
    </a>
  </div>,
  <main
    className="Main"
  >
    <section
      className="Toolbar"
    >
      <section
        className="Header"
      >
        <img
          alt="yuhur_img"
          src="profile.jpg"
        />
        <h1>
          Yurii Hurianov
        </h1>
        <h5 />
      </section>
      <section
        className="InfoArea"
      >
        <h5>
          Languages
        </h5>
        <hr />
        <h5>
          Contact info
        </h5>
        <hr />
        <section
          className="socials"
        >
          <a
            aria-label="test"
            className="social-icon"
            href="test"
            style={
              Object {
                "display": "inline-block",
                "height": "50px",
                "overflow": "hidden",
                "position": "relative",
                "verticalAlign": "middle",
                "width": "50px",
              }
            }
          >
            <div
              className="social-container"
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <svg
                className="social-svg"
                style={
                  Object {
                    "borderRadius": "50%",
                    "fillRule": "evenodd",
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
                viewBox="0 0 64 64"
              >
                <g
                  className="social-svg-background"
                  style={
                    Object {
                      "MozTransition": "fill 170ms ease-in-out",
                      "OTransition": "fill 170ms ease-in-out",
                      "WebkitTransition": "fill 170ms ease-in-out",
                      "fill": "transparent",
                      "msTransition": "fill 170ms ease-in-out",
                      "transition": "fill 170ms ease-in-out",
                    }
                  }
                >
                  <circle
                    cx="32"
                    cy="32"
                    r="31"
                  />
                </g>
                <g
                  className="social-svg-icon"
                  style={
                    Object {
                      "MozTransition": "fill 170ms ease-in-out",
                      "OTransition": "fill 170ms ease-in-out",
                      "WebkitTransition": "fill 170ms ease-in-out",
                      "fill": "transparent",
                      "msTransition": "fill 170ms ease-in-out",
                      "transition": "fill 170ms ease-in-out",
                    }
                  }
                >
                  <path
                    d={null}
                  />
                </g>
                <g
                  className="social-svg-mask"
                  style={
                    Object {
                      "MozTransition": "fill 170ms ease-in-out",
                      "OTransition": "fill 170ms ease-in-out",
                      "WebkitTransition": "fill 170ms ease-in-out",
                      "fill": null,
                      "msTransition": "fill 170ms ease-in-out",
                      "transition": "fill 170ms ease-in-out",
                    }
                  }
                >
                  <path
                    d={null}
                  />
                </g>
              </svg>
            </div>
          </a>
        </section>
      </section>
      <footer
        className="footer"
      >
        <span>
          Â© copyright
        </span>
        <a
          href="https://github.com/YuraGB/newProject"
        >
          This project is on the Github
        </a>
      </footer>
    </section>
  </main>,
]
`;
